core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -670,7 +671,9 @@ public static void pingService(InetSocketAddress address, alluxio.grpc.ServiceTy
         .build();
     try {
       ServiceVersionClientServiceGrpc.ServiceVersionClientServiceBlockingStub versionClient =
-          ServiceVersionClientServiceGrpc.newBlockingStub(channel);
+          ServiceVersionClientServiceGrpc.newBlockingStub(channel)
+              .withDeadlineAfter(conf.getMs(PropertyKey.WORKER_MASTER_PERIODICAL_CONNECT_TIMEOUT),
```

Comment:
`pingService` is used in a bunch of places including client and job services. We probably want to use a different configuration property. Can we maybe reuse the `USER_MASTER_POLLING_TIMEOUT` or something more generic?